### DISCUSSION TITLE
Run all unit tests after running a single test

### DISCUSSION THREAD

**Discussion by joshuafolkken** [MikeSchulze/gdUnit4#562] [Q&A] [answered]
I'm using the Mac Option+F5 shortcut to test a specific function. After doing this, when I want to run all tests and press only F5, it still runs the test for just the previous function.

I've found that to make F5 run all tests, I need to first run Option+F5 on a blank line.

Is this the expected behavior?

I like GdUnit4. Thank you!

## Environment Details

- GdUnit4 4.3.4
- Godot 4.3, stable, Typed GDScript
- macOS Sonoma 14.6.1

**Comment by MikeSchulze** [comment, marked as answer]
The test can be run in different scenarios.
At first, you have the overall shortcuts:
![image](https://github.com/user-attachments/assets/d1c11b4e-0374-4a16-aef6-5fd081610e3e)

F5 only should not run any test, exceptional you have not built a custom main scene to run tests.

Or using the context menus in the script editor or inspector.
The test execution is depended on the actual selected line in the script editor.
e.g. you have selected the test function, it will run only this single test, otherwise all tests in the test suite will be executed.
In the inspector the same behavior it will execute the selected test, test suite or the content of a selected folder.

I hope this answers your questions.

**Reply by joshuafolkken** [reply]
Thank you!
